src/smtml/expr.ml

```diff
@@ -423,6 +461,11 @@ let rec relop ty op hte1 hte2 =
   | Ne, _, Val (App (`Op "symbol", [ Str _ ]))
   | Ne, Val (App (`Op "symbol", [ Str _ ])), _ ->
     value True
+  | Eq, (
+    (Symbol ({ty=Ty_fp prec1; _} as s1)) as e1),
+    (Symbol ({ty=Ty_fp prec2; _} as s2))
+       when prec1 = prec2 && Symbol.equal s1 s2 ->
+     make (Unop (Ty_bool, Not, make (Unop (Ty_fp prec1, Is_nan, make e1))))
```

Comment:
```suggestion
     raw_unop Ty_bool Not (raw_unop (Ty_fp prec1) Is_nan hte1)
```

Uses the `raw_unop` operator and `hte1` instead of `e1` to doing a hashcons lookup